### PR TITLE
Simplify detection of errors for execute_reply messages

### PR DESCRIPTION
**Pull request by takluyver**
Closes gh-143

I forget when we introduced ExecutionResult to IPython, but this requires a version of IPython with that machinery.

### CODE REVIEW

File: ipykernel/ipkernel.py [hunk 1]
```diff
@@ -348,25 +342,23 @@ def do_apply(self, content, bufs, msg_id, reply_metadata):
...
-                e_info = dict(engine_uuid=self.ident, engine_id=self.int_id, method='apply')
-                reply_content['engine_info'] = e_info
+            reply_content = {
+                u'traceback': shell._last_traceback or [],
+                u'ename': unicode_type(type(e).__name__),
+                u'evalue': safe_unicode(e),
+            }
+            # FIXME: deprecate piece for ipyparallel:
+            e_info = dict(engine_uuid=self.ident, engine_id=self.int_id, method='apply')
+            reply_content['engine_info'] = e_info
```
Comment (takluyver) [inline ipykernel/ipkernel.py]: @minrk this is in the handler for `apply` messages for ipyparallel - can you check if it makes sense?

Comment (minrk) [inline ipykernel/ipkernel.py]: Yeah, should be fine

**Comment by minrk** [comment]
Looks like ExecutionResult is in 4.0, which is already the minimum supported version.

**Comment by minrk** [comment]
Nice! I didn't know about ExecutionResult.

**Comment by minrk** [comment]
I didn't think about the fact that ipyparallel would have its own copy of the `_reply_content` hack, since the `apply` implementation in ipykernel is all deprecated. This means that ipykernel 4.4 won't work with ipyparallel ∈ [5,5.2). I don't think it's a huge deal, as it's a shared hack and I'll release ipyparallel this week, too. Just another reason to do away with the hack altogether.

**Pull request merged**